src/main/java/eu/lundegaard/liferay/db/setup/core/SetupFragments.java

```diff
@@ -0,0 +1,252 @@
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * @author michal.volf@lundegaard.eu
+ */
+public class SetupFragments {
+
+    private static final Log LOG = LogFactoryUtil.getLog(SetupFragments.class);
+
+    private SetupFragments() {}
+
+    public static void setupFragments(List<FragmentCollection> fragmentCollections, long userId, long groupId) {
+        for (FragmentCollection fragmentCollection : fragmentCollections) {
+            ServiceContext serviceContext = new ServiceContext();
+
+            switch (fragmentCollection.getSetupAction()) {
+                case "create":
+                    createFragmentCollection(fragmentCollection, userId, groupId, serviceContext);
+                    break;
+                case "update":
+                    updateFragmentCollection(fragmentCollection, groupId);
+                    break;
+                case "delete":
+                    deleteFragmentCollection(fragmentCollection, groupId);
+                    break;
+                default:
+                    throw new IllegalStateException("Illegal setup action " + fragmentCollection.getSetupAction());
+            }
+
+            LOG.info("Setup for fragment collection " + fragmentCollection.getName() + " finished");
+        }
+    }
+
+    private static void createFragmentCollection(FragmentCollection fragmentCollection, long userId, long groupId,
+            ServiceContext serviceContext) {
+        String collectionName = fragmentCollection.getName();
+        LOG.info("Creating fragment collection " + collectionName);
+
+        try {
+            Optional<com.liferay.fragment.model.FragmentCollection> existingCollection =
+                    findFragmentCollection(collectionName, groupId);
+
+            if (existingCollection.isPresent()) {
+                LOG.info("Fragment collection " + collectionName + " already exists, skipping...");
+            } else {
+                com.liferay.fragment.model.FragmentCollection createdCollection =
+                        FragmentCollectionLocalServiceUtil.addFragmentCollection(
+                                userId, groupId, collectionName, fragmentCollection.getDescription(),
+                                serviceContext);
+
+                for (Fragment fragment : fragmentCollection.getFragment()) {
+                    switch (fragment.getSetupAction()) {
+                        case "create":
+                            createFragment(fragment, userId, groupId, createdCollection, serviceContext);
+                            break;
+                        case "update":
+                            updateFragment(fragment, userId, groupId, createdCollection);
+                            break;
+                        case "delete":
+                            deleteFragment(fragment, groupId, createdCollection);
+                            break;
+                        default:
+                            throw new IllegalStateException(
```

Comment:
`IllegalStateException` = component is in wrong state, you want `IllegalArgumentException`